sdk/Sdk/CustomAttributeExtensions.cs

```diff
@@ -14,16 +14,40 @@ public static class CustomAttributeExtensions
         public static IDictionary<string, object> GetAllDefinedProperties(this CustomAttribute attribute)
         {
             var properties = new Dictionary<string, object>();
-
             // To avoid needing to instantiate any types, assume that the constructor
             // argument names are equal to property names.
+            LoadDefaultProperties(properties, attribute);
             LoadConstructorArguments(properties, attribute);
             LoadDefinedProperties(properties, attribute);
 
             return properties;
         }
 
-        private static void LoadConstructorArguments(IDictionary<string, object> properties, CustomAttribute attribute)
+        private static IEnumerable<(string, CustomAttributeArgument?)> GetDefaultValues(this CustomAttribute attribute)
```

Comment:
this is all should be removed later... after above todo is done: `avoid needing to instantiate any types, assume that the constructor argument names are equal to property names.`